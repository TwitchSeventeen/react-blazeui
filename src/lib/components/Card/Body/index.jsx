import React from 'react';
import PropTypes from 'prop-types';

function Body (props) {
  const { children } = props;
  return (
      <div className="c-card__body">
        { children }
      </div>
  );
}

Body.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Body.defaultProps = {
  children: null,
};

export default Body;