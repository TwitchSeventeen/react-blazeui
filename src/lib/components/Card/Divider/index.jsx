import React from 'react';
import PropTypes from 'prop-types';

function Divider (props) {
  const { children } = props;
  return (
      <div role="separator" className="c-card__item c-card__item--divider">
        { children }
      </div>
  );
}

Divider.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Divider.defaultProps = {
  children: null,
};

export default Divider;