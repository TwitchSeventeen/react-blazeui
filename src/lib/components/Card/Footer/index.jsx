import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Footer(props) {
  const { children, block, ...other } = props;
  const classes = classNames('c-card__footer', { 'c-card__footer--block': block });
  return (
    <footer {...other} className={classes}>
      { children }
    </footer>
  );
}

Footer.propTypes = {
  /**
   * A block footer with no padding
   */
  block: PropTypes.bool,
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Footer.defaultProps = {
  block: false,
  children: null,
};

export default Footer;
