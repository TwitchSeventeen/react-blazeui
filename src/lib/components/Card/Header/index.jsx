import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Header(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('c-card__header', className);
  return (
    <header
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </header>
  );
}

Header.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Header.defaultProps = {
  children: null,
  className: null,
  dataName: 'card-header',
};

export default Header;
