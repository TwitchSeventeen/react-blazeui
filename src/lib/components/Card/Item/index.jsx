import React from 'react';
import PropTypes from 'prop-types';

function Item (props) {
  const { children } = props;
  return (
      <div className="c-card__item">
        { children }
      </div>
  );
}

Item.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Item.defaultProps = {
  children: null,
};

export default Item;