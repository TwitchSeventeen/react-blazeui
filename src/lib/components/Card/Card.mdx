---
name: Card
---

import { Playground, PropsTable } from 'docz'
import Card from './index';

# Card

Cards are a common minimalistic design tool used to segregate sections of your content.

## Properties
### Card
<PropsTable of={Card} />

### Card.Body
<PropsTable of={Card.Body} />

### Card.Item
<PropsTable of={Card.Item} />

### Card.Divider
<PropsTable of={Card.Divider} />

## Basic Card
<Playground>
    <Card>
        <Card.Body>
            <p>Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.</p>
        </Card.Body>
    </Card>
</Playground>

## Dividers
<Playground>
    <Card>
        <Card.Divider>
            Header
        </Card.Divider>
        <Card.Item>
            <p>Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.</p>
            <p>Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.</p>
        </Card.Item>
        <Card.Divider>
            Divider
        </Card.Divider>
        <Card.Item>
            <p>Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.</p>
            <p>Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.</p>
        </Card.Item>
        <Card.Divider>
            Footer
        </Card.Divider>
    </Card>
</Playground>