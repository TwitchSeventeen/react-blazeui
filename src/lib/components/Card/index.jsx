import React from 'react';
import PropTypes from 'prop-types';
import Body from './Body';
import Divider from './Divider';
import Item from './Item';

function Card (props) {
  const { children } = props;
  return (
      <div className="c-card">
        { children }
      </div>
  );
}

Card.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Card.defaultProps = {
  children: null,
};

Card.Body = Body;
Card.Divider = Divider;
Card.Item = Item;

export default Card;