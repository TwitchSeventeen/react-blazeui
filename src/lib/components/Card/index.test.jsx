import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from './index';

describe('Card', () => {
  it('should be defined', () => {
    expect(Card).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Card>
          <p>lorem</p>
        </Card>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});