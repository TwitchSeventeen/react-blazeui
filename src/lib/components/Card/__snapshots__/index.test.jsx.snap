// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="c-card"
>
  <p>
    lorem
  </p>
</div>
`;
